composer.json

```diff
@@ -80,25 +80,31 @@
         "sort-packages": true
     },
     "scripts": {
+        "post-autoload-dump": [
+            "@install-tools"
+        ],
         "cs:check": "@php php-cs-fixer fix --dry-run --diff",
         "cs:fix": "@php php-cs-fixer fix",
-        "dev-tools:check": [
-            "./dev-tools/check_file_permissions.sh",
-            "./dev-tools/check_trailing_spaces.sh"
-        ],
-        "dev-tools:docs": "@php dev-tools/doc.php",
-        "dev-tools:install": "@composer -d dev-tools install",
-        "docs": "@dev-tools:docs",
+        "docs": "@php dev-tools/doc.php",
+        "install-tools": "@composer -d dev-tools install",
         "mess-detector": "@php dev-tools/vendor/bin/phpmd . ansi phpmd.xml",
-        "phpstan": "@php -d memory_limit=256M dev-tools/vendor/bin/phpstan analyse",
-        "phpstan:baseline": "@php -d memory_limit=256M dev-tools/vendor/bin/phpstan analyse --generate-baseline=./dev-tools/phpstan/baseline.php",
+        "normalize": [
+            "@composer normalize -d dev-tools --dry-run ../composer.json",
+            "@composer normalize -d dev-tools --dry-run composer.json"
+        ],
+        "phpstan": "@php -d memory_limit=512M dev-tools/vendor/bin/phpstan analyse",
+        "phpstan:baseline": "@php -d memory_limit=512M dev-tools/vendor/bin/phpstan analyse --generate-baseline=./dev-tools/phpstan/baseline.php",
         "qa": "@quality-assurance",
         "quality-assurance": [
             "Composer\\Config::disableProcessTimeout",
             "@sa",
             "@test"
         ],
         "sa": "@static-analysis",
+        "self-check": [
+            "./dev-tools/check_file_permissions.sh",
+            "./dev-tools/check_trailing_spaces.sh"
+        ],
         "static-analysis": [
```

Comment:
i would like to have a command that runs all, Fixer, stan, file permissions, mess detector, ...

Reply:
I will add more things to `qa`, but it's impossible to include Mess Detector there because of:
- [this](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/6839#discussion_r1289266913): deprecation warnings can be fixed by updating `pdepend/pdepend` (I will do it in separate PR), but problem with [allowed modifiers](https://github.com/phpmd/phpmd/issues/506) remains. Analysing `src` and `tests` works ok, issue is somewhere in `dev-tools` but I wasn't able to get `--exclude` working.
- Mess Detector in CI is run against `$CHANGED_PHP_FILES` and running it on whole codebase returns many violations. Adding it to Composer script would cause fail for every run and it does not make sense.

Reply:
`dev-tools/check_trailing_spaces.sh` also can't be added to `qa` because it fails for me and prints false-negatives like:

```
tests/UtilsTest.php:343:     * @dataProvider provideNaturalLanguageJoinWithBackticksCases
tests/UtilsTest.php:407:     * @dataProvider provideToStringCases
tests/UtilsTest.php:53:     * @dataProvider provideCamelCaseToUnderscoreCases
tests/WhitespacesFixerConfigTest.php:29:     * @dataProvider provideFixCases
tests/WordMatcherTest.php:29:     * @param string[] $candidates
tests/WordMatcherTest.php:31:     * @dataProvider provideMatchCases
```

It looks like it finds lines ending with `s` rather than with space (`\s`):

<img width="1251" alt="image" src="https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/assets/600668/83dbbfe4-dd9f-403b-9c57-e0f980af29ab">

We can add it when we fix it, in separate PR.

Reply:
OS incompatibility?

Reply:
@keradus see #7216, as I wrote before I'll add it here after merging that one and rebasing here.

Reply:
@keradus I've prepare improvements for Mess Detector in #7224, `mess-detector` script was already added to `qa` but will fail - we need to merge that PR first to achieve "clean start" for violations.

Reply:
7224 merged ;)